lib/timers.js

```diff
@@ -203,6 +214,14 @@ function clearTimeout(timer) {
   if (timer && timer._onTimeout) {
     timer._onTimeout = null;
     unenroll(timer);
+    return;
+  }
+  if (typeof timer === 'number' || typeof timer === 'string') {
```

Comment:
IIRC `typeof` checks are really fast so this check could probably come first and then re-use the same logic?

Reply:
Thanks, will benchmark that.

Reply:
Local testing shows a bit worse results with more regressions, I think 2 line code duplication is fine:
```cmd
                                                              confidence improvement accuracy (*)   (**)   (***)
 timers/immediate.js type='breadth' n=5000000                                -0.08 %       ±6.26% ±8.38% ±11.03%
 timers/immediate.js type='breadth1' n=5000000                               -1.88 %       ±4.07% ±5.46%  ±7.19%
 timers/immediate.js type='breadth4' n=5000000                               -2.46 %       ±3.94% ±5.29%  ±6.99%
 timers/immediate.js type='clear' n=5000000                                   0.71 %       ±2.64% ±3.53%  ±4.65%
 timers/immediate.js type='depth' n=5000000                                   0.25 %       ±3.03% ±4.06%  ±5.34%
 timers/immediate.js type='depth1' n=5000000                                  1.19 %       ±3.33% ±4.46%  ±5.87%
 timers/set-immediate-breadth-args.js n=5000000                               1.33 %       ±1.93% ±2.59%  ±3.44%
 timers/set-immediate-breadth.js n=10000000                                  -1.51 %       ±4.29% ±5.75%  ±7.56%
 timers/set-immediate-depth-args.js n=5000000                                -0.25 %       ±0.97% ±1.30%  ±1.71%
 timers/timers-breadth-args.js n=1000000                                     -0.27 %       ±3.63% ±4.86%  ±6.40%
 timers/timers-breadth.js n=5000000                                          -3.15 %       ±3.51% ±4.71%  ±6.19%
 timers/timers-cancel-pooled.js n=5000000                             **      4.34 %       ±2.62% ±3.52%  ±4.65%
 timers/timers-cancel-unpooled.js direction='end' n=1000000           **     -3.13 %       ±2.00% ±2.68%  ±3.53%
 timers/timers-cancel-unpooled.js direction='start' n=1000000        ***     -2.77 %       ±0.94% ±1.26%  ±1.66%
 timers/timers-depth.js n=1000                                                0.03 %       ±0.77% ±1.03%  ±1.36%
 timers/timers-insert-pooled.js n=5000000                                    -1.01 %       ±2.39% ±3.20%  ±4.21%
 timers/timers-insert-unpooled.js direction='end' n=1000000                   0.39 %       ±1.12% ±1.50%  ±1.97%
 timers/timers-insert-unpooled.js direction='start' n=1000000                -0.09 %       ±1.90% ±2.55%  ±3.36%
 timers/timers-timeout-nexttick.js n=50000                             *      5.10 %       ±4.72% ±6.35%  ±8.38%
 timers/timers-timeout-nexttick.js n=5000000                                 -2.12 %       ±2.26% ±3.03%  ±4.00%
 timers/timers-timeout-pooled.js n=10000000                                  -1.05 %       ±1.61% ±2.15%  ±2.83%
 timers/timers-timeout-unpooled.js n=1000000                                 -2.04 %       ±3.44% ±4.61%  ±6.07%

Be aware that when doing many comparisons the risk of a false-positive
result increases. In this case there are 22 comparisons, you can thus
expect the following amount of false-positive results:
  1.10 false positives, when considering a   5% risk acceptance (*, **, ***),
  0.22 false positives, when considering a   1% risk acceptance (**, ***),
  0.02 false positives, when considering a 0.1% risk acceptance (***)
```